tests/testthat/test-class-spec.R

```diff
@@ -49,14 +54,58 @@ test_that("can work with S4 classes", {
   klass <- methods::getClass("Range")
 
   expect_equal(class_type(klass), "s4")
-  expect_equal(class_names(klass), "Range")
+  expect_equal(class_dispatch(klass), "Range")
+  expect_equal(class_register(klass), "Range")
   expect_s4_class(class_construct(klass, start = 1, end = 2), "Range")
   expect_equal(class_desc(klass), "S4<Range>")
   expect_equal(class_deparse(klass), "Range")
 
   obj <- methods::new(klass, start = 1, end = 1)
   expect_equal(obj_type(obj), "s4")
   expect_equal(obj_desc(obj), "S4<Range>")
+  expect_equal(obj_dispatch(obj), "Range")
+  expect_equal(class_inherits(obj, klass), TRUE)
+})
+
+test_that("can work with S4 subclasses", {
+  methods::setClass("Foo1", slots = c(start = "numeric", end = "numeric"))
+  methods::setClass("Foo2", contains = "Foo1")
+  klass <- methods::getClass("Foo2")
+
+  expect_equal(class_type(klass), "s4")
+  expect_equal(class_dispatch(klass), c("Foo2", "Foo1"))
+  expect_equal(class_register(klass), "Foo2")
```

Comment:
Are there cases where `class_register(x) != class_dispatch(x)[[1]]`? Besides the `NULL` case? I was looking for a test where that was the case.